src/dimmwitted.cc

```diff
@@ -165,14 +273,17 @@ void DimmWitted::learn() {
 
   double current_stepsize = opts.stepsize;
   const std::unique_ptr<double[]> prev_weights(new double[nweight]);
+  std::vector<float> delta(nweight);  // 32-bit float
```

Comment:
The main reason for using vector is that `msgpack` seems to support that only. Folks on SO also seem to recommend using vectors from some of my random searches... It's fixed size, so perf should be an issue.